### PR TITLE
Makefile improvements

### PR DESCRIPTION
**Pull request by ryandesign**
Here are some Makefile improvements I developed when I added displayplacer to MacPorts:

### Use `$CC` instead of `gcc`

Use `$CC` instead of `gcc`. There isn't a real `gcc` on macOS; it's a link to `clang`. The user may wish to use a different compiler specified in `$CC`. If not, `make` defaults to `cc`, which is also a link to `clang`.

### Use `$@` and `$<`

Use `$@` for the target and `$<` for the first prerequisite rather than repeating that information in the rule.

### Use `$CFLAGS`, `$CPPFLAGS`, and `$LDFLAGS`

Use `$CFLAGS`, `$CPPFLAGS`, and `$LDFLAGS` that the user may have specified. Add `-I.` first to ensure that the local directory is searched for include files first, before any `-I` paths that might be specified in `$CFLAGS` or `$CPPFLAGS`.

### Add an "all" target

Add an "all" target. Users familiar with automake may expect to be able to run `make all`.

### Add a "clean" target

Add a "clean" target that removes what was built.

### Add an "install" target

Add an "install" target that installs the program and license file to a typical prefix, defaulting to /usr/local but overridable with standard variables.

The use of `mkdir -p` here is not portable but works on macOS which is the only OS this Makefile is expected to be used with.

### CODE REVIEW

**Comment by ryandesign** [comment]
Updated the "install" target to also install the license file.

**Review by rbradcurtis — APPROVED** [review]
Reviewed, builds clean on MacOS Monterey 12.4.  Recommend for merge.

**Comment by jakehilborn** [comment]
Thank you for the contribution! I added also added a debug target and most likely did it in a non-standard way.

File: Makefile [hunk 1]
```diff
@@ -1,8 +1,4 @@
-prefix ?= /usr/local
-exec_prefix ?= $(prefix)
-bindir ?= $(exec_prefix)/bin
-datarootdir ?= $(prefix)/share
-docdir ?= $(datarootdir)/doc/displayplacer
+bindir ?= /usr/local/bin
```
Comment (ryandesign) [inline Makefile:1]: It wasn't boilerplate. `prefix` and `exec_prefix` are standard autotools variables users may wish to override. And your `install` target should install your LICENSE file.

**Pull request merged**